pkg/agent/controller/networkpolicy/packetin.go

```diff
@@ -327,3 +333,88 @@ func (c *Controller) rejectRequest(pktIn *ofctrl.PacketIn) error {
 			true)
 	}
 }
+
+func (c *Controller) storeDenyConnection(pktIn *ofctrl.PacketIn) error {
+	packet, err := binding.ParsePacketIn(pktIn)
+	if err != nil {
+		return fmt.Errorf("error in parsing packetin: %v", err)
+	}
+
+	// Get 5-tuple information
+	flowKey := flowexporter.Tuple{
+		SourcePort:      packet.SourcePort,
+		DestinationPort: packet.DestinationPort,
+		Protocol:        packet.IPProto,
+	}
+	if packet.IsIPv6 {
+		flowKey.SourceAddress = packet.SourceIP.To16()
+		flowKey.DestinationAddress = packet.DestinationIP.To16()
+	} else {
+		flowKey.SourceAddress = packet.SourceIP.To4()
+		flowKey.DestinationAddress = packet.DestinationIP.To4()
+	}
+
+	// Generate deny connection and add to deny connection store
+	denyConn := flowexporter.Connection{}
+	denyConn.FlowKey = flowKey
+
+	// No need to obtain connection info again if it already exists in denyConnectionStore.
+	if conn, exist := c.denyConnStore.GetConnByKey(flowexporter.NewConnectionKey(&denyConn)); exist {
+		conn.DeltaBytes = uint64(packet.IPLength)
+		conn.StopTime = time.Now()
+		c.denyConnStore.AddOrUpdateConn(conn)
```

Comment:
`GetConnByKey` gets `*flowexporter.Connection` in the cache, and it seems to me that the code will override its value, I guess `DeltaBytes` will not be accumulated as expected in `AddOrUpdateConn`. I think this can be confirmed by an unit test.

Reply:
Updated AddOrUpdateConn for deny connections to avoid override. Thanks!